landmark preparation still not interrupted if failed

Related to #107 and #980

It happily does the preprocessing further for the other profiles. Is this due to the error and not just an exception?

```
2017-05-11 06:42:48,403 [fastest_racingbike] ERROR com.graphhopper.routing.lm.LMAlgoFactoryDecorator - Problem while LM preparation fastest_racingbike
com.carrotsearch.hppc.BufferAllocationException: Not enough memory to allocate buffers for rehashing: 134,217,728 -> 268,435,456
        at com.carrotsearch.hppc.IntObjectHashMap.allocateBuffers(IntObjectHashMap.java:1109)
        at com.carrotsearch.hppc.IntObjectHashMap.allocateThenInsertThenRehash(IntObjectHashMap.java:1134)
        at com.carrotsearch.hppc.IntObjectHashMap.put(IntObjectHashMap.java:179)
        at com.graphhopper.routing.DijkstraBidirectionRef.fillEdges(DijkstraBidirectionRef.java:175)
        at com.graphhopper.routing.DijkstraBidirectionRef.fillEdgesFrom(DijkstraBidirectionRef.java:132)
        at com.graphhopper.routing.AbstractBidirAlgo.runAlgo(AbstractBidirAlgo.java:68)
        at com.graphhopper.routing.lm.LandmarkStorage$LandmarkExplorer.runAlgo(LandmarkStorage.java:748)
        at com.graphhopper.routing.lm.LandmarkStorage.createLandmarksForSubnetwork(LandmarkStorage.java:364)
        at com.graphhopper.routing.lm.LandmarkStorage.createLandmarks(LandmarkStorage.java:278)
        at com.graphhopper.routing.lm.PrepareLandmarks.doWork(PrepareLandmarks.java:130)
        at com.graphhopper.routing.lm.LMAlgoFactoryDecorator$1.run(LMAlgoFactoryDecorator.java:286)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.OutOfMemoryError: Java heap space
        at com.carrotsearch.hppc.IntObjectHashMap.allocateBuffers(IntObjectHashMap.java:1100)
        ... 17 more
2017-05-11 06:44:46,161 [fastest_foot] ERROR com.graphhopper.routing.lm.LMAlgoFactoryDecorator - Problem while LM preparation fastest_foot
com.carrotsearch.hppc.BufferAllocationException: Not enough memory to allocate buffers for rehashing: 134,217,728 -> 268,435,456
        at com.carrotsearch.hppc.IntObjectHashMap.allocateBuffers(IntObjectHashMap.java:1109)
        at com.carrotsearch.hppc.IntObjectHashMap.allocateThenInsertThenRehash(IntObjectHashMap.java:1134)
        at com.carrotsearch.hppc.IntObjectHashMap.put(IntObjectHashMap.java:179)
        at com.graphhopper.routing.DijkstraBidirectionRef.fillEdges(DijkstraBidirectionRef.java:175)
        at com.graphhopper.routing.DijkstraBidirectionRef.fillEdgesFrom(DijkstraBidirectionRef.java:132)
        at com.graphhopper.routing.AbstractBidirAlgo.runAlgo(AbstractBidirAlgo.java:68)
        at com.graphhopper.routing.lm.LandmarkStorage$LandmarkExplorer.runAlgo(LandmarkStorage.java:748)
        at com.graphhopper.routing.lm.LandmarkStorage.createLandmarksForSubnetwork(LandmarkStorage.java:364)
        at com.graphhopper.routing.lm.LandmarkStorage.createLandmarks(LandmarkStorage.java:278)
        at com.graphhopper.routing.lm.PrepareLandmarks.doWork(PrepareLandmarks.java:130)
        at com.graphhopper.routing.lm.LMAlgoFactoryDecorator$1.run(LMAlgoFactoryDecorator.java:286)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.OutOfMemoryError: Java heap space
        at com.carrotsearch.hppc.IntObjectHashMap.allocateBuffers(IntObjectHashMap.java:1101)
        ... 17 more
2017-05-11 06:47:51,418 [fastest_bike2] INFO  com.graphhopper.routing.lm.LandmarkStorage - Finding landmarks [fastest|bike2] in network [106422030]. Progress 0%, totalMB:163840, usedMB:13387
2017-05-11 06:47:54,283 [fastest_hike] INFO  com.graphhopper.routing.lm.LandmarkStorage - Finding la...
```

Furthermore the required RAM seems to be too high for the prepratation (25-35GB RAM per profile). Investigate this.

--- a/core/src/main/java/com/graphhopper/routing/ch/CHAlgoFactoryDecorator.java
+++ b/core/src/main/java/com/graphhopper/routing/ch/CHAlgoFactoryDecorator.java
@@ -286,19 +286,10 @@ public class CHAlgoFactoryDecorator implements RoutingAlgorithmFactoryDecorator 
             completionService.submit(new Runnable() {
                 @Override
                 public void run() {
-                    String errorKey = CH.PREPARE + "error." + name;
-                    try {
-                        // toString is not taken into account so we need to cheat, see http://stackoverflow.com/q/6113746/194609 for other options
-                        Thread.currentThread().setName(name);
-                        properties.put(errorKey, "CH preparation incomplete");
-                        prepare.doWork();
-                        properties.remove(errorKey);
-                        properties.put(CH.PREPARE + "date." + name, Helper.createFormatter().format(new Date()));
-                    } catch (Exception ex) {
-                        LOGGER.error("Problem while CH preparation " + name, ex);
-                        properties.put(errorKey, ex.getMessage());
-                        throw ex;
-                    }
+                    // toString is not taken into account so we need to cheat, see http://stackoverflow.com/q/6113746/194609 for other options
+                    Thread.currentThread().setName(name);
+                    prepare.doWork();
+                    properties.put(CH.PREPARE + "date." + name, Helper.createFormatter().format(new Date()));
                 }
             }, name);
 

--- a/core/src/main/java/com/graphhopper/routing/lm/LMAlgoFactoryDecorator.java
+++ b/core/src/main/java/com/graphhopper/routing/lm/LMAlgoFactoryDecorator.java
@@ -279,23 +279,16 @@ public class LMAlgoFactoryDecorator implements RoutingAlgorithmFactoryDecorator 
             completionService.submit(new Runnable() {
                 @Override
                 public void run() {
-                    String errorKey = Landmark.PREPARE + "error." + name;
-                    try {
-                        Thread.currentThread().setName(name);
-                        properties.put(errorKey, "LM preparation incomplete");
-                        plm.doWork();
-                        properties.remove(errorKey);
-                        properties.put(Landmark.PREPARE + "date." + name, Helper.createFormatter().format(new Date()));
-                    } catch (Exception ex) {
-                        LOGGER.error("Problem while LM preparation " + name, ex);
-                        properties.put(errorKey, ex.getMessage());
-                    }
+                    Thread.currentThread().setName(name);
+                    plm.doWork();
+                    properties.put(Landmark.PREPARE + "date." + name, Helper.createFormatter().format(new Date()));
                 }
             }, name);
             submittedPreparations++;
         }
 
         threadPool.shutdown();
+
         try {
             for (int i = 0; i < submittedPreparations; i++) {
                 completionService.take().get();

